R/summary_funs.R

```diff
@@ -406,92 +438,120 @@ get_stat <- function(mu, lppd, y_wobs_test, stat, mu.bs = NULL, lppd.bs = NULL,
           rep(1L, y[i_short]))
       }))
       mu <- rep(mu, y_wobs_test$wobs)
-      if (!is.null(mu.bs)) {
-        mu.bs <- rep(mu.bs, y_wobs_test$wobs)
-        # CAUTION: If `y` is allowed to have `NA`s here, then `n_notna.bs` needs
-        # to be adapted:
-        n_notna.bs <- sum(!is.na(mu.bs))
-      }
-      # CAUTION: If `y` is allowed to have `NA`s here, then `n_notna` needs to
-      # be adapted:
-      n_notna <- sum(!is.na(mu))
-      if (!is.null(n_notna.bs) &&
-          getOption("projpred.additional_checks", FALSE)) {
-        stopifnot(n_notna == n_notna.bs)
+      if (!is.null(summaries_baseline)) {
+        mu_baseline <- rep(summaries_baseline$mu, y_wobs_test$wobs)
+      } else {
+        mu_baseline <- NULL
       }
       wcv <- rep(wcv, y_wobs_test$wobs)
-      wcv <- n_notna * wcv / sum(wcv)
+      wcv <- n * wcv / sum(wcv)
     } else {
       stopifnot(all(y_wobs_test$wobs == 1))
+      if (!is.null(summaries_baseline)) {
+        mu_baseline <- summaries_baseline$mu
+      } else {
+        mu_baseline <- NULL
+      }
     }
     if (stat %in% c("acc", "pctcorr")) {
       # Find out whether each observation was classified correctly or not:
       if (!is.factor(mu)) {
         mu <- round(mu)
       }
-      crrct <- mu == y
+      correct <- mu == y
 
-      if (!is.null(mu.bs)) {
-        if (!is.factor(mu.bs)) {
-          mu.bs <- round(mu.bs)
+      if (!is.null(mu_baseline)) {
+        if (!is.factor(mu_baseline)) {
+          mu_baseline <- round(mu_baseline)
         }
-        crrct.bs <- mu.bs == y
+        correct_baseline <- mu_baseline == y
+      } else {
+        correct_baseline <- 0
+      }
 
-        value <- mean(wcv * (crrct - crrct.bs), na.rm = TRUE)
-        value.se <- weighted.sd(crrct - crrct.bs, wcv, na.rm = TRUE) /
-          sqrt(n_notna)
+      if (!is.null(summaries_fast) && sum(n_loo<n)) {
+        # subsampling difference estimator (Magnusson et al., 2020)
+        mu_fast <- summaries_fast$mu
+        if (!is.factor(mu_fast)) {
+          mu_fast <- round(mu_fast)
+        }
+        correct_fast <- mu_fast == y
+        srs_diffe <- .srs_diff_est_w(y_approx = correct_fast - correct_baseline,
+                                     y = (correct-correct_baseline)[loo_inds],
+                                     y_idx = loo_inds,
+                                     w = wcv)
+        value <- srs_diffe$y_hat / n + mean(wcv * correct_baseline)
+        # combine estimates of var(y_hat) and var(y)
+        value_se <- sqrt(srs_diffe$v_y_hat + srs_diffe$hat_v_y) / n
       } else {
-        value <- mean(wcv * crrct, na.rm = TRUE)
-        value.se <- weighted.sd(crrct, wcv, na.rm = TRUE) / sqrt(n_notna)
+        # full LOO estimator
+        value <- mean(wcv * correct)
```

Comment:
Isn't `- correct_baseline` missing here (would need to be added in parentheses)?

Reply:
I saw you added ` - correct_baseline`, giving now `value <- mean(wcv * correct - correct_baseline)`, but I think we need to add parentheses to have `value <- mean(wcv * (correct - correct_baseline))`, right?

Reply:
I should not have added that as this is connected to showing all other criteria except **pd in original scale

Reply:
Ok, then this part of the code is correct, under the assumption that you want to keep the new `deltas = TRUE` behavior.